### ISSUE TITLE
Feature Request: Helper method to support for all Smart Contract data types

### ISSUE THREAD

**Issue by piyalbasu** [stellar/js-soroban-client#43]
In trying to build a Transaction with a Contract operation with params, I found it difficult to convert JS primitives to certain data types (for ex: `u64`). We will need to support all possible data types used in smart contracts

Additionally, I found the existing pattern of creating SC Values to be a bit hard to follow and I suspect other FE eng's will have a similar issue. For ex: `xdr.ScVal.scvObject(
      xdr.ScObject.scoVec([xdr.ScVal.scvSymbol('Invoker')])
    )`
    
We will ideally have helpers akin to `numberToU64`, etc. I see similar helper are found in `@soroban-react/utils` but this should be provided natively by this lib.

**Comment by dmkozh** [comment]
I would also suggest trying to look at this problem a bit more holistically and try providing a way to define the schema and parse/fill the arguments according to it. The motivation is that currently most of the data on contract side has schema (normally it's a Rust enum/struct/value serialized to `ScVal`).

**Comment by Shaptic** [comment]
Moving this to stellar-base where it belongs!
